e2e/metrics/metrics_test.go

```diff
@@ -50,6 +50,7 @@ func TestMetrics(t *testing.T) {
 	cluster3.Establish(t,
```

Comment:
Might be nice to (be able to) show state both before and after the test runs.

Either this new `ShowState()` option could do that (perhaps optionally), or...

If `Establish()` returned the `Cluster` it makes, and `dump` was exported instead of private, then one could dump state whenever they'd like, including after the test:

```go
c := cluster3.Establish(t, ...)
c.Dump() // before
t.Cleanup(c.Dump) // after
```

or similar.
Might even be nice to include when any test using this helper fails?  Could add a Cleanup that checks if `t.Failed()` and auto-dump.  That might produce real noisy output though, idk.

Reply:
Yeah I thought about doing something like that. What we _really_ want is a way to take a snapshot of the cluster state before a test run, then dump that snapshot after the test fails (and maybe also dump the state of the cluster after the test (or even better, a diff)). Left as an exercise for the reader, of course.